src/utils/walletConnect.ts

```diff
@@ -84,64 +151,88 @@ export const signClient = Promise.resolve(
 function parseRPCParams({
   method,
   params,
-}: {
-  method: string;
-  params: string[];
-}) {
-  if (method === 'eth_sign' || method === 'personal_sign') {
-    const [address, message] = params.sort(a =>
-      ethersUtils.isAddress(a) ? -1 : 1
-    );
-    const isHexString = ethersUtils.isHexString(message);
+}: RPCPayload): {
+  address?: string;
+  message?: string;
+} {
+  switch (method) {
+    case 'eth_sign':
+    case 'personal_sign': {
+      const [address, message] = params.sort(a =>
+        ethersUtils.isAddress(a) ? -1 : 1
+      );
+      const isHexString = ethersUtils.isHexString(message);
 
-    const decodedMessage = isHexString
-      ? ethersUtils.toUtf8String(message)
-      : message;
+      const decodedMessage = isHexString
+        ? ethersUtils.toUtf8String(message)
+        : message;
 
-    return {
-      address,
-      message: decodedMessage,
-    };
+      return {
+        address,
+        message: decodedMessage,
+      };
+    }
+    /**
+     * @see https://eips.ethereum.org/EIPS/eip-712#specification-of-the-eth_signtypeddata-json-rpc
+     * @see https://docs.metamask.io/guide/signing-data.html#a-brief-history
+     */
+    case 'eth_signTypedData':
+    case 'eth_signTypedData_v1':
+    case 'eth_signTypedData_v3':
+    case 'eth_signTypedData_v4': {
+      const [address, message] = params;
+
+      return {
+        address,
+        message: JSON.parse(message),
+      };
+    }
+    case 'eth_sendTransaction': {
+      const [tx] = params;
+      return {
+        address: tx.from,
+      };
+    }
+    default:
+      return {};
   }
+}
 
-  if (method === 'eth_signTypedData' || method === 'eth_signTypedData_v4') {
-    const [address, message] = params.sort(a =>
-      ethersUtils.isAddress(a) ? -1 : 1
-    );
+export function isSupportedSigningMethod(method: RPCMethod) {
+  return [
+    RPCMethod.Sign,
+    RPCMethod.PersonalSign,
+    RPCMethod.SignTypedData,
+    RPCMethod.SignTypedDataV1,
+    RPCMethod.SignTypedDataV3,
+    RPCMethod.SignTypedDataV4,
+  ].includes(method);
+}
 
-    return {
-      address,
-      message: JSON.parse(message),
-    };
-  }
+export function isSupportedTransactionMethod(method: RPCMethod) {
+  return [RPCMethod.SendTransaction].includes(method);
+}
 
-  return {};
+export function isSupportedMethod(method: RPCMethod) {
+  return (
+    isSupportedSigningMethod(method) || isSupportedTransactionMethod(method)
+  );
```

Comment:
Re-implemented these because we changed what we support. I put them here because there isn't a better place to put them atm afaik.